FirebaseSegmentation/Sources/SEGNetworkManager.m

```diff
@@ -0,0 +1,223 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "FirebaseSegmentation/Sources/SEGNetworkManager.h"
+
+#import "FirebaseCore/Sources/Private/FIRAppInternal.h"
+#import "FirebaseCore/Sources/Private/FIRLogger.h"
+#import "FirebaseCore/Sources/Private/FIROptionsInternal.h"
+
+// TODO(dmandar): define in build file.
+#define SEG_ALPHA_SERVER
+
+static NSString *const kServerURLDomain = @"https://firebasesegmentation.googleapis.com";
+
+#ifdef SEG_ALPHA_SERVER
+static NSString *const kServerURLVersion = @"/v1alpha";
+#else
+static NSString *const kServerURLVersion = @"/v1";
+#endif
+
+static NSString *const kServerURLStringProjects = @"/projects/";
+static NSString *const kServerURLStringInstallations = @"/installations/";
+static NSString *const kServerURLStringCustomSegmentationData = @"/customSegmentationData";
+
+static NSString *const kHTTPMethodPatch = @"PATCH";
+static NSString *const kRequestHeaderAuthorizationValueString = @"FIREBASE_INSTALLATIONS_AUTH";
+static NSString *const kRequestDataCustomInstallationIdString = @"custom_installation_id";
+
+// HTTP header names.
+static NSString *const kHeaderNameAPIKey = @"x-goog-api-key";
+static NSString *const kHeaderNameFirebaseAuthorizationToken = @"Authorization";
+static NSString *const kHeaderNameContentType = @"Content-Type";
+static NSString *const kHeaderNameContentEncoding = @"Content-Encoding";
+static NSString *const kHeaderNameAcceptEncoding = @"Accept-Encoding";
+
+// Sends the bundle ID. Refer to b/130301479 for details.
+static NSString *const kiOSBundleIdentifierHeaderName =
+    @"X-Ios-Bundle-Identifier";  ///< HTTP Header Field Name
+
+/// Config HTTP request content type JSON
+static NSString *const kContentTypeValueJSON = @"application/json";
+
+// TODO: Handle error codes.
+/// HTTP status codes. Ref: https://cloud.google.com/apis/design/errors#error_retries
+static NSInteger const kSEGResponseHTTPStatusCodeOK = 200;
+// static NSInteger const kSEGResponseHTTPStatusCodeConflict = 409;
+// static NSInteger const kSEGResponseHTTPStatusTooManyRequests = 429;
+// static NSInteger const kSEGResponseHTTPStatusCodeInternalError = 500;
+// static NSInteger const kSEGResponseHTTPStatusCodeServiceUnavailable = 503;
+// static NSInteger const kSEGResponseHTTPStatusCodeGatewayTimeout = 504;
+
+// HTTP default timeout.
+static NSTimeInterval const kSEGHTTPRequestTimeout = 60;
+
+/// Completion handler invoked by URLSession completion handler.
+typedef void (^URLSessionCompletion)(NSData *data, NSURLResponse *response, NSError *error);
+
+@implementation SEGNetworkManager {
+  FIROptions *_firebaseAppOptions;
+  NSURLSession *_URLSession;
+}
+
+- initWithFIROptions:(FIROptions *)options {
```

Comment:
This method is missing a return type. Also, the `FIR` prefix doesn't need to be in the method selector:

```objc
- (instancetype)initWithOptions:(FIROptions *)options {
  // ...
}
```